staging/src/k8s.io/apiextensions-apiserver/examples/client-go/pkg/client/informers/externalversions/factory.go

```diff
@@ -75,6 +75,14 @@ func WithNamespace(namespace string) SharedInformerOption {
 	}
 }
 
+// WithCustomInformer sets a custom informer for the given informer type.
+func WithCustomInformer(obj v1.Object, informer cache.SharedIndexInformer) SharedInformerOption {
+	return func(factory *sharedInformerFactory) *sharedInformerFactory {
+		factory.informers[reflect.TypeOf(obj)] = informer
+		return factory
+	}
+}
+
```

Comment:
can't the same be done already today?

```
factory.InformerFor(obj, func(versioned.Interface, time.Duration) cache.SharedIndexInformer {
  return informer
})
```

Reply:
Not that calling `InformerFor` is praticularly nice. I actually prefer the option func style.

Reply:
@sttts you are right, I have overlooked that. so.. do you think it is worhwhile adding this option func?

Reply:
Thanks @sttts. That perfectly fits my requirement.

I will leave it to you to decide whether to continue the "WithCustomInformer" idea or not.